### PR TITLE
[toolchain] Added `--dest-dir` switch

### LINKED ISSUE #4219: SW uses toolchain in 2 places
I have set `TOOLCHAIN_PATH`
> echo $TOOLCHAIN_PATH
> /edascratch/ms_edatools/riscv-tools/riscv-gnu-toolchain/20200904-1

And then find meson uses the toolchain as I set in `$TOOLCHAIN_PATH`, but [toolchain.txt](https://github.com/lowRISC/opentitan/blob/master/toolchain.txt) is hard-coded to use the default one `/tools/riscv/`

So, I got this log from simulation. Can we change it to all use $TOOLCHAIN_PATH if it's set?

> Using meson toolchain file at **/edascratch/ms_edatools/riscv-tools/riscv-gnu-toolchain/20200904-1/meson-riscv32-unknown-elf-clang.txt**.
> \+ meson -Dot_version=opentitan-snapshot-20191101-1-3141-g9e163d8c -Dbin_dir=/edascratch/weicai-opentitan/ot/top_earlgrey.sim.vcs/master/sw/build-bin -Dtock_local=false -Dkeep_includes=false -Dcoverage=false --cross-file=/edascratch/ms_edatools/riscv-tools/riscv-gnu-toolchain/20200904-1/meson-riscv32-unknown-elf-clang.txt /edascratch/weicai-opentitan/ot/top_earlgrey.sim.vcs/master/sw/build-out
> Purging superfluous -I arguments from /edascratch/weicai-opentitan/ot/top_earlgrey.sim.vcs/master/sw/build-out.
> The Meson build system
> Version: 0.54.0
> Source dir: /usr/local/google/home/weicai/ot/uart
> Build dir: /edascratch/weicai-opentitan/ot/top_earlgrey.sim.vcs/master/sw/build-out
> Build type: cross build
> WARNING: Unknown options: "build.c_std, build.cpp_std"
> The value of new options can be set with:
> meson setup <builddir> --reconfigure -Dnew_option=new_value ...
> Project name: opentitan
> Project version: 0.1
> C compiler for the build machine: cc (gcc 10.2.0 "cc (Debian 10.2.0-9) 10.2.0")
> C linker for the build machine: cc ld.bfd 2.35.1
> C++ compiler for the build machine: c++ (gcc 10.2.0 "c++ (Debian 10.2.0-9) 10.2.0")
> C++ linker for the build machine: c++ ld.bfd 2.35.1
> C compiler for the host machine: **/tools/riscv/bin/riscv32-unknown-elf-clang** (clang 11.0.0 "lowRISC clang version 11.0.0 (https://github.com/llvm/llvm-project.git fa038e03504c7d0dfd438b1dfdd6da7081e75617)")

@sriyerg is working on fixing this.

### PR DESCRIPTION
**Pull request by sriyerg**
See first commit for the list of changes. The second commit just performs a lint cleanup. 

- Fixes #4219

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Comment by lenary** [comment]
LGTM! 

I'll do a follow-up which makes `--target-dir` default to the environment variable, if it's set.

**Review by lenary — APPROVED** [review]

File: util/get-toolchain.py [hunk 1]
```diff
@@ -115,9 +115,15 @@ def install(archive_file, target_dir):
...
-    """This rewrites the toolchain configuration files to point to install_path"""
-    if str(install_path) == TARGET_DIR:
+def postinstall_rewrite_configs(install_path, release_path):
+    """Rewrites the toolchain configuration files to point to release_path.
+
+    In most cases, the install_path and release_path are identical. However, in
+    corporate environments with centrally installed tools, the installation may
+    be done in a temporary staging area first (by the engineers who need it for
+    their work), before it is moved to the release area (managed by a separate
+    IT department)."""
```
Comment (imphil) [inline util/get-toolchain.py:125]: Can you make this explanation less of an example/motivation, but more of a description of what the paths actually are? E.g. install_path is the final installation directory from which the tools are called, release_path is a installation staging directory, from where the tools are expected to be manually moved to install_dir.

File: util/get-toolchain.py [hunk 2]
```diff
@@ -138,6 +144,16 @@ def main():
...
+    parser.add_argument('--release-dir',
+                        required=False,
+                        help="""Release directory, if different from target
+                        directory. If not set, this defaults to the target
+                        directory. Useful in corporate environments where a
+                        more global tool installation area is preferred, but
+                        due to IT restrictions, the installation has to be done
+                        in a temporary staging area first. The tool version
+                        to be installed is appended to this path by default.
+                        """)
```
Comment (imphil) [inline util/get-toolchain.py:156]: I find the naming and explanation rather confusing now. Can we do two things:

- Switch to a more common naming, such as
  * `install-dir`: the final installation path
  * `dest-dir` (or `install-staging-dir`): the installation staging directory, from where it is moved to the final install-dir.

  See https://www.gnu.org/software/automake/manual/html_node/Staged-Installs.html for the automake documentation on it.
* Explain what these directories do not by example. You can also remove the "corporate IT environment" motivation from the help text.

File: doc/ug/install_instructions/index.md [hunk 3]
```diff
@@ -149,8 +149,9 @@ $ cd $REPO_TOP
 $ ./util/get-toolchain.py
 ```
 
-This tool will automatically adjust the toolchain configuration if you override the installation directory (by using the `--target-dir` option).
-Alternatively, manually download the file starting with `lowrisc-toolchain-rv32imc-` from [GitHub releases](https://github.com/lowRISC/lowrisc-toolchains/releases/latest) and unpack it to `/tools/riscv`.
+This tool will automatically adjust the toolchain configuration if you override the installation directory (by using the `-install-dir` option).
```
Comment (imphil) [inline doc/ug/install_instructions/index.md:152]:
```suggestion
This tool will automatically adjust the toolchain configuration if you override the installation directory (by using the `--install-dir` option).
```

**Review by imphil — APPROVED** [review]
Thanks a lot Sri, LGTM!

**Pull request merged**